crates/astria-proto/src/native/composer/block_relay/v1alpha1/server.rs

```diff
@@ -0,0 +1,81 @@
+use std::{
+    collections::HashMap,
+    hash::Hash,
+};
+
+use tokio::sync::{
+    mpsc,
+    oneshot,
+};
+use tonic::Response;
+
+use crate::{
+    generated::block_relay::v1alpha1::{
+        top_of_block_relay_server::TopOfBlockRelay,
+        GetBidRequest,
+        GetBidResponse,
+        GetBundleRequest,
+        GetBundleResponse,
+    },
+    native::{
+        composer::block_relay::v1alpha1::bid::SignedBundle,
+        sequencer::v1alpha1::Address,
+    },
+};
+
+pub(super) struct TopOfBlockRelayService {
+    // Channel to request the current best bid from the top of block relay actor
+    // TODO: needs a better name
+    best_bid_request_tx: mpsc::Sender<oneshot::Sender<SignedBundle>>,
+    // Bids that have been committed to by proposers
+    pending_commitment: HashMap<Address, SignedBundle>,
```

Comment:
A few problems I have with this:
1. Should this state be kept here or should this object communicate via channels to the service object (in the composer crate)?
2. I need to keep the `SignedBundles` that are waiting for a commitment since the commitment will need to be verified against them. If I map it from address, and the `GetBidRequest` has an address field, anyone can call this with a fake address and override the proposer's pending state. Am I overthinking this? Should `GetBidRequest` require the proposer to sign over the block height for which they want the bundle?